packages/promotion/src/models/promotion.ts

```diff
@@ -0,0 +1,13 @@
+import { generateEntityId } from "@medusajs/utils"
+import { BeforeCreate, Entity, PrimaryKey } from "@mikro-orm/core"
+
+@Entity()
+export default class Promotion {
+  @PrimaryKey({ columnType: "text" })
+  id!: string
+
+  @BeforeCreate()
```

Comment:
todo: add oninit hook as well as we discussed in the previous modules but has not been done yet except in the search pr which is not merged yet.

```ts
 @OnInit()
  async onInit() {
    this.id = generateEntityId(this.id, "promo")
  }
```